report_qweb_table_pagebreak/models/ir_qweb.py

```diff
@@ -0,0 +1,104 @@
+from lxml import html
+
+from odoo import api, models
+
+from odoo.addons.base.models.qweb import QWeb
+
+
+class StyleParser:
+    """Class to manipulate and Serialize/DeSerialize style attribute on
+    html tag::
+
+        style = StyleParser(element.get("style"))
+        # set attribute page-break-inside
+        style.set("page-break-inside", "avoid")
+        # Serialize style to set style value on html element
+        element.set("style", str(style))
+    """
+
+    _styles: dict = {}
+
+    def __init__(self, style_content: str):
+        self._parse(style_content)
+
+    def _parse(self, style_content):
+        self._styles = {}
+        for style in style_content.split(";"):
+            data_style = style.split(":")
+            if len(data_style) != 2:
+                # ignore if not expected string
+                continue
+            key, value = data_style
+            self._styles[key.strip()] = value.strip()
+
+    def __str__(self):
+        return "; ".join([f"{k}: {v}" for k, v in self._styles.items()])
+
+    def set(self, key, value, default=None):
+        if value is None:
+            value = default
+        if value is None:
+            self._styles.pop(key, None)
+        else:
+            self._styles[key] = value
+
+
+class IrQWeb(models.AbstractModel, QWeb):
+    _inherit = "ir.qweb"
+    _description = "Qweb"
+
+    @api.model
+    def _render(self, id_or_xml_id, values=None, **options):
+        result = super()._render(id_or_xml_id, values=values, **options)
+
+        if b"data-page-break=" not in result:
```

Comment:
```suggestion
        if b"data-page-break-inside=" not in result:
```
might be more readable: `data-page-break-inside="avoid"` vs `data-page-break="always"`.
This would also be more similar to the attribute `page-break-inside` used under the hood.
What do you think?

Reply:
I prefer to use only one attribute such as `qweb-page-break` or `qweb-table-page-break` because:

* it keeps things different from css attributes (code is using `page-break-inside` and `page-break-after`) and avoid to mix-up both
* I feel easier to use only one attribute and manage behavior according the value 
* In qweb template that would manage the value I suppose code will be also easier to manage

I'll add a comment that any other value (thant `avoid`, `before` or `after`) are ignored